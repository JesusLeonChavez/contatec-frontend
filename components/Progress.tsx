import { Progress, Flex, Text, Grid, Box } from "@chakra-ui/react"
import ZIcon from "./Icon/ZIcon"
interface ProgressProps {
  value: number
  start: string
  quantity: number
}
export default function ProgressValoration({
  value,
  start,
  quantity
}: ProgressProps) {
  return (
    <Grid templateColumns="80% 20%" py="1">
      <Progress colorScheme="yellow" size="lg" value={value} borderRadius="5" />
      <Flex align="center" justify="center">
        <Box d="flex" justifyContent="start" w="100px">
          <Text fontSize="xs" px="1">
            {start}
          </Text>
          <ZIcon name="star" color="secondary" />
          <Text fontSize="xs" px="3">
            ({quantity})
          </Text>
        </Box>
      </Flex>
    </Grid>
  )
}
