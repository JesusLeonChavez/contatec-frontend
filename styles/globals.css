:root{
  --primary: #706F6F;
  --secondary: #304659;
}

html,
body {
  padding: 0;
  margin: 0;

}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
  font-family: 'Poppins', sans-serif;
}

.generalWrapper{
  max-width: 1300px;
  margin: 0 auto;
  overflow: auto;
  padding: 0 40px;
}

.flexCenterCustom{
  display: flex;
  align-items: center;
  justify-content: center;
}


/*----------------Swiper-----------------------*/
.swiper-container {
  width: 100%;
  height: 100%;
}

.swiper-slide {
  text-align: center;
  font-size: 18px;
  background: red;
  /* Center slide text vertically */
  display: -webkit-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-align-items: center;
  align-items: center;
}

.slide-2{
  background: green;
}

.mySwiper  .swiper-button-next, .mySwiper  .swiper-button-prev {
  color: white;
  border: 1px solid white;
  border-radius: 50%;
  padding: 1.3rem;
}

.mySwiper .swiper-button-next{
  top: 800px;
  right: 100px;
}

.mySwiper .swiper-button-prev{
  top: 800px;
  right: 180px;
  left: auto;
}

.mySwiper .swiper-button-next::after,.mySwiper .swiper-button-prev::after{
  font-size: 1rem;
}


.mySwiper2{
  width: 200px;
}
.mySwiper2 .swiper-container{
  width: 100%;
  height: 100%;
}
.mySwiper2 .swiper-slide{
  text-align: center;
  font-size: 18px;
  background: #fff;
  background: pink;
  /* Center slide text vertically */
  display: -webkit-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-align-items: center;
  align-items: center;
}
.mySwiper2 .slide-2{
  background: blue;
}


.mySwiper3{
  width:100%;
}
.mySwiper3 .swiper-container{
  width: 300px;
  height: 100%;
}
.mySwiper3 .swiper-slide{
  text-align: center;
  font-size: 18px;
  background: #fff;
  background: purple;
  height: 500px;
  /* Center slide text vertically */
  display: -webkit-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-align-items: center;
  align-items: center;
}
.mySwiper3 .slide-2{
  background: blue;
}

.mySwiper3 .swiper-button-next, .mySwiper3 .swiper-button-prev {
  color: white;
  border: 1px solid white;
  border-radius: 50%;
  padding: 1.3rem;
}

.mySwiper3 .swiper-button-next{
  top: 450px;
  right: 100px;
}
.mySwiper3 .swiper-button-prev{
  top: 450px;
  right: 180px;
  left: auto;
}

.mySwiper3 .swiper-button-next::after, .mySwiper3 .swiper-button-prev::after{
  font-size: 1rem;
}




