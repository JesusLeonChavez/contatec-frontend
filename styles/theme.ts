import { extendTheme } from '@chakra-ui/react'
import { ButtonStyles as Button } from './components/buttonStyles'
export const newTheme = extendTheme({
  colors: {
    primary: '#EB0029',
    secondary: '#FF6F91',
    highlight: '#00C9A7'
  },
  components: {
    Button
  }
})
