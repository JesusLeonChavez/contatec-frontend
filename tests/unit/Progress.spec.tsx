/**

 * @jest-environment jsdom

 */

import { render } from "@testing-library/react"

import "@testing-library/jest-dom"

import { newTheme } from "../../styles/theme"

import { ChakraProvider } from "@chakra-ui/react"

import Progress from "../../components/Progress"

import React from "react"

const ChakraRenderer = ({ children }) => {
  return <ChakraProvider theme={newTheme}>{children}</ChakraProvider>
}

describe("Progress", function () {
  let expectedProps

  beforeEach(() => {
    expectedProps = {
      value: 100,
      start: "3",
      quantity: 12
    }
  })

  test("debe renderizarse", () => {
    const component = render(<Progress {...expectedProps} />, {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      wrapper: ChakraRenderer
    })

    expect(component).toMatchSnapshot()
  })

  test("debe recibir las propiedades con el tipo esperado", () => {
    expect(typeof expectedProps.value).toBe("number")
    expect(typeof expectedProps.start).toBe("string")
    expect(typeof expectedProps.quantity).toBe("number")
  })

  test("debe renderizar start, name y quantity", () => {
    const { getByText } = render(<Progress {...expectedProps} />, {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      wrapper: ChakraRenderer
    })

    const start = getByText(expectedProps.start)
    const quantity = getByText(`(${expectedProps.quantity})`)

    expect(start).toBeVisible()

    expect(quantity).toBeVisible()
  })
})
