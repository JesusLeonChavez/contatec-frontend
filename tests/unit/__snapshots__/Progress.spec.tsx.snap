// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress debe renderizarse 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-2ypazg"
      >
        <div
          class="css-177remz"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="100"
            class="css-l8gkht"
            role="progressbar"
            style="width: 100%;"
          />
        </div>
        <div
          class="css-gmuwbf"
        >
          <div
            class="css-1r7zza8"
          >
            <p
              class="chakra-text css-np8v3k"
            >
              3
            </p>
            <svg
              color="var(--secondary)"
              fill="none"
              height="1em"
              style="font-size: 15px;"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M3.5 8.695l4.547-.702L9.998 3.8l1.952 4.193 4.55.702-3.3 3.418.755 4.687-3.957-2.197L6.043 16.8l.754-4.687L3.5 8.695z"
                fill="var(--secondary)"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M10 15.925L4.438 19 5.5 12.488 1 7.875l6.219-.95L10 1l2.781 5.925 6.219.95-4.5 4.613L15.562 19 10 15.925zM4.12 8.92l2.984 3.063-.683 4.199L10 14.214l3.58 1.968-.683-4.2 2.985-3.062-4.116-.63L10 4.534 8.234 8.29l-4.113.63z"
                fill="#FBD76D"
                fill-rule="evenodd"
              />
            </svg>
            <p
              class="chakra-text css-ipe0h1"
            >
              (
              12
              )
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-2ypazg"
    >
      <div
        class="css-177remz"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="100"
          class="css-l8gkht"
          role="progressbar"
          style="width: 100%;"
        />
      </div>
      <div
        class="css-gmuwbf"
      >
        <div
          class="css-1r7zza8"
        >
          <p
            class="chakra-text css-np8v3k"
          >
            3
          </p>
          <svg
            color="var(--secondary)"
            fill="none"
            height="1em"
            style="font-size: 15px;"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M3.5 8.695l4.547-.702L9.998 3.8l1.952 4.193 4.55.702-3.3 3.418.755 4.687-3.957-2.197L6.043 16.8l.754-4.687L3.5 8.695z"
              fill="var(--secondary)"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M10 15.925L4.438 19 5.5 12.488 1 7.875l6.219-.95L10 1l2.781 5.925 6.219.95-4.5 4.613L15.562 19 10 15.925zM4.12 8.92l2.984 3.063-.683 4.199L10 14.214l3.58 1.968-.683-4.2 2.985-3.062-4.116-.63L10 4.534 8.234 8.29l-4.113.63z"
              fill="#FBD76D"
              fill-rule="evenodd"
            />
          </svg>
          <p
            class="chakra-text css-ipe0h1"
          >
            (
            12
            )
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
